File: 2026/array/max_square_by_removing_fence_2975.py
class Solution:
    def maximizeSquareArea(
        self, m: int, n: int, hFences: List[int], vFences: List[int]
    ) -> int:
        M = int(1e9 + 7)

        hFences.append(1)
        hFences.append(m)

        vFences.append(1)
        vFences.append(n)

        hFences.sort()
        vFences.sort()

        widths = set()
        for i in range(len(vFences)):
            for j in range(i + 1, len(vFences)):
                width = vFences[j] - vFences[i]
                widths.add(width)

        maxSide = 0
        for i in range(len(hFences)):
            for j in range(i + 1, len(hFences)):
                height = hFences[j] - hFences[i]
                if height in widths:
                    maxSide = max(maxSide, height)

        return -1 if maxSide == 0 else (maxSide * maxSide) % M

File: 2026/array/max_square_by_removing_fence_2975.js
var maximizeSquareArea = function (m, n, hFences, vFences) {
    const MOD = 1_000_000_007;

    hFences.push(1, m);
    vFences.push(1, n);

    hFences.sort((a, b) => a - b);
    vFences.sort((a, b) => a - b);

    // Store all vertical differences
    const vDiffs = new Set();
    for (let i = 0; i < vFences.length; i++) {
        for (let j = i + 1; j < vFences.length; j++) {
            vDiffs.add(vFences[j] - vFences[i]);
        }
    }

    let maxSide = 0;

    // Try horizontal differences (larger first)
    for (let i = 0; i < hFences.length; i++) {
        for (let j = i + 1; j < hFences.length; j++) {
            const diff = hFences[j] - hFences[i];
            if (vDiffs.has(diff)) {
                maxSide = Math.max(maxSide, diff);
            }
        }
    }

    if (maxSide === 0) return -1;

    return Number((BigInt(maxSide) * BigInt(maxSide)) % BigInt(MOD));
};
